core/src/main/java/com/amazon/opendistroforelasticsearch/sql/expression/operator/predicate/UnaryPredicateOperator.java

```diff
@@ -100,6 +99,25 @@ private static FunctionResolver isNotNull() {
                 Collectors.toList()));
   }
 
+  private static FunctionResolver ifFunction() {
+    FunctionName functionName = BuiltinFunctionName.IF.getName();
+    List<ExprCoreType> typeList = ExprCoreType.coreTypes();
+
+    List<SerializableFunction<FunctionName, org.apache.commons.lang3.tuple.Pair<FunctionSignature,
+            FunctionBuilder>>> functionsOne = typeList.stream().map(v ->
+            impl((UnaryPredicateOperator::exprIf), v, BOOLEAN, v, v))
+            .collect(Collectors.toList());
+
+    List<SerializableFunction<FunctionName, org.apache.commons.lang3.tuple.Pair<FunctionSignature,
+            FunctionBuilder>>> functionsTwo = typeList.stream().map(v ->
+            impl((UnaryPredicateOperator::exprIf), v, UNKNOWN, v, v))
+            .collect(Collectors.toList());
```

Comment:
Same as my previous question for line 130-133, why is this required?

Reply:
could we consider the solution which could for case...when and if. i think they both control flow statement which may not fit into our function defintion.

Reply:
Good idea.